packages/portfolio-deploy/test/portfolio.integration.test.ts

```diff
@@ -0,0 +1,201 @@
+import { makeNodeBundleCache } from '@endo/bundle-source/cache.js';
+import { execCmd, checkPodsRunning } from '../scripts/shell.ts';
+import childProcess from 'node:child_process';
+import { createRequire } from 'node:module';
+import fsp from 'node:fs/promises';
+import { commonSetup, ensureAccounts } from './support.js';
+import { makeContainer } from '../tools/agd-lib.js';
+
+const nodeRequire = createRequire(import.meta.url);
+const configFile = nodeRequire.resolve(
+  '../../../multichain-testing/config.ymax.yaml',
+);
+
+const contractName = 'ymax0';
+const chainInfoBuilder = './src/chain-info.build.js';
+const portfolioBuilder = './src/portfolio.build.js';
+
+interface TestContext {
+  tools: {
+    deployBuilder: (...args: any[]) => Promise<any>;
+    retryUntilCondition: (...args: any[]) => Promise<any>;
+    vstorageClient: {
+      queryData: (path: string) => Promise<any>;
+    };
+    agd?: any;
+  };
+  deployBuilder: (...args: any[]) => Promise<any>;
+  retryUntilCondition: (...args: any[]) => Promise<any>;
+  wallets: any;
+}
+
+const test: TestFn<TestContext> = anyTest;
+
+const fundWallets = async (t: ExecutionContext, wallets: any) => {
+  const addr = wallets.gov1;
+  console.log('Funding wallet:', addr);
+  try {
+    console.log('\n🔄 Initiating USDC IBC transfer from Noble → Agoric...');
+    const usdcTransferCmd = [
+      `kubectl exec -i noblelocal-genesis-0 -- nobled tx ibc-transfer transfer transfer channel-0`,
+      `${addr}`,
+      `10000000uusdc`,
+      `--from test1`,
+      `--keyring-backend test`,
+      `--chain-id noblelocal`,
+      `--fees 2000uusdc`,
+      `--gas auto`,
+      `--gas-adjustment 1.3`,
+      `-y`,
+    ].join(' ');
+
+    const usdcResult = await execCmd(usdcTransferCmd);
+    console.log('✅ USDC transfer command output:\n', usdcResult);
+    t.assert(
+      usdcResult.includes('code: 0') || usdcResult.includes('txhash'),
+      'USDC transfer failed',
+    );
+
+    console.log('\n💸 Sending BLD + IST from Agoric genesis...');
+    const bankSendCmd = [
+      `kubectl exec -i agoriclocal-genesis-0 -- agd tx bank send`,
+      `genesis`,
+      `${addr}`,
+      `500000000000ubld,500000000000uist`,
+      `--keyring-backend=test`,
+      `--chain-id=agoriclocal`,
+      `--fees=2000ubld`,
+      `--gas=auto`,
+      `--gas-adjustment=1.3`,
+      `-y`,
+    ].join(' ');
+
+    const bankResult = await execCmd(bankSendCmd);
+    console.log('✅ BLD/IST bank send output:\n', bankResult);
+    t.assert(
+      bankResult.includes('code: 0') || bankResult.includes('txhash'),
+      'Bank send failed',
+    );
+
+    console.log('\n⏳ Waiting for transactions to finalize...');
+    await new Promise(r => setTimeout(r, 5000));
+
+    console.log('\n🔍 Querying balances from agoriclocal...');
+    const queryCmd = `kubectl exec -i agoriclocal-genesis-0 -- agd query bank balances ${addr} --output json`;
+    const queryOutput = await execCmd(queryCmd);
```

Comment:
Now we don't need to think about `kubectl` any more:

```suggestion
    const queryOutput = await agd.withFlags({ output: 'json').exec(`query bank balances ${addr}`.split(' '));
```

That assumes `$addr` has no spaces. execa would take care of that if we used it as a template tag, but the `const queryCmd = ...` line uses plain string templates.

`makeCmdRunner` doesn't (yet) provide a template tag to take care of shell quoting, but it's straightforward to be robust in this case:

```suggestion
    const queryOutput = await agd.withFlags({ output: 'json').exec(['query', 'bank', 'balances', addr]);
```